helpers/prism.js

```diff
@@ -0,0 +1,432 @@
+	lineNumbersSeparator: '#1b1b1b',
+	tokenDefault: '#1b1b1b',
+	tokenComment: '#9e9e9e',
+	tokenPunctuation: '#9e9e9e',
+	tokenNumber: '#007936',
+	tokenUrl: '#9e9e9e',
+	tokenOperator: '#9e9e9e',
+	tokenInterpolation: '#872bff',
+	tokenAttributeName: '#d30038',
+	tokenConstant: '#007936',
+	tokenProperty: '#d30038',
+	tokenTag: '#007936',
+	tokenBoolean: '#007936',
+	tokenEntity: 'inherit',
+	tokenInterpolationPunctuation: '#872bff',
+	tokenFunction: '#d30038',
+	tokenClassName: '#d30038',
+	tokenSymbol: '#007936',
+	tokenBuiltin: '#872bff',
+	tokenKeyword: '#0069c2',
+	tokenAtRule: '#007936',
+	tokenSelector: '#872bff',
+	tokenImportant: 'inherit',
+	tokenRegex: 'inherit',
+	tokenString: '#007936',
+	tokenChar: '#007936',
+	tokenAttributeValue: '#007936'
+});
+
+export const getCodeColors = mode => {
+	return mode === colorModes.DARK ? darkColors : lightColors;
+};
+
+const generateColorVariables = mode => {
+	const colors = getCodeColors(mode);
+	return `
+		--d2l-code-background: ${colors.background};
+		--d2l-code-language: ${colors.language};
+		--d2l-code-line-numbers: ${colors.lineNumbers};
+		--d2l-code-line-numbers-separator: ${colors.lineNumbersSeparator};
+		--d2l-code-token-atrule: ${colors.tokenAtRule};
+		--d2l-code-token-attribute-name: ${colors.tokenAttributeName};
+		--d2l-code-token-attribute-value: ${colors.tokenAttributeValue};
+		--d2l-code-token-boolean: ${colors.tokenBoolean};
+		--d2l-code-token-builtin: ${colors.tokenBuiltin};
+		--d2l-code-token-char: ${colors.tokenChar};
+		--d2l-code-token-class-name: ${colors.tokenClassName};
+		--d2l-code-token-comment: ${colors.tokenComment};
+		--d2l-code-token-constant: ${colors.tokenConstant};
+		--d2l-code-token-default: ${colors.tokenDefault};
+		--d2l-code-token-entity: ${colors.tokenEntity};
+		--d2l-code-token-function: ${colors.tokenFunction};
+		--d2l-code-token-important: ${colors.tokenImportant};
+		--d2l-code-token-interpolation: ${colors.tokenInterpolation};
+		--d2l-code-token-interpolation-punctuation: ${colors.tokenInterpolationPunctuation};
+		--d2l-code-token-keyword: ${colors.tokenKeyword};
+		--d2l-code-token-number: ${colors.tokenNumber};
+		--d2l-code-token-operator: ${colors.tokenOperator};
+		--d2l-code-token-property: ${colors.tokenProperty};
+		--d2l-code-token-punctuation: ${colors.tokenPunctuation};
+		--d2l-code-token-regex: ${colors.tokenRegex};
+		--d2l-code-token-selector: ${colors.tokenSelector};
+		--d2l-code-token-string: ${colors.tokenString};
+		--d2l-code-token-symbol: ${colors.tokenSymbol};
+		--d2l-code-token-tag: ${colors.tokenTag};
+		--d2l-code-token-url: ${colors.tokenUrl};
+	`;
+};
+
+export const codeStyles = css`
+
+	pre[class*="language-"].d2l-code,
+	code[class*="language-"].d2l-code {
+		${unsafeCSS(generateColorVariables(colorModes.LIGHT))}
+	}
+
+	pre[class*="language-"].d2l-code-dark,
+	code[class*="language-"].d2l-code-dark {
+		${unsafeCSS(generateColorVariables(colorModes.DARK))}
+	}
+
+	pre[class*="language-"].d2l-code,
```

Comment:
Note: many of these styles are "standard" across Prism themes. We need to be careful to preserve specificity in cases where another tool (ex. tinyMCE) happens to include Prism styles, or a content author has referenced their own version of Prism.